mslice/plotting/pyplot.py

```diff
@@ -411,49 +411,11 @@ def disconnect(cid):
     return get_current_fig_manager().canvas.mpl_disconnect(cid)
 
 
-def close(*args):
+def close(num):
     """
     Close a figure window.
-
-    ``close()`` by itself closes the current figure
-
-    ``close(fig)`` closes the `.Figure` instance *fig*
-
-    ``close(num)`` closes the figure number *num*
-
-    ``close(name)`` where *name* is a string, closes figure with that label
-
-    ``close('all')`` closes all the figure windows
     """
-
-    if len(args) == 0:
-        figManager = GlobalFigureManager.get_active()
-        if figManager is None:
-            return
-        else:
-            GlobalFigureManager.destroy(figManager.num)
-    elif len(args) == 1:
-        arg = args[0]
-        if arg == 'all':
-            GlobalFigureManager.destroy_all()
-        elif isinstance(arg, int):
-            GlobalFigureManager.destroy(arg)
-        elif hasattr(arg, 'int'):
-            # if we are dealing with a type UUID, we
-            # can use its integer representation
-            GlobalFigureManager.destroy(arg.int)
-        elif isinstance(arg, str):
-            allLabels = get_figlabels()
-            if arg in allLabels:
-                num = get_fignums()[allLabels.index(arg)]
-                GlobalFigureManager.destroy(num)
-        elif isinstance(arg, Figure):
-            GlobalFigureManager.destroy_fig(arg)
-        else:
-            raise TypeError('Unrecognized argument type %s to close'
-                            % type(arg))
-    else:
-        raise TypeError('close takes 0 or 1 arguments')
+    GlobalFigureManager.figure_closed(num)
```

Comment:
Out of curiosity is there a reason we need to deviate from the standard `matplotlib` behaviour?

Reply:
I tried calling the method that was here before and I got this:
```
File "C:\Users\ygh95688\PycharmProjects\mslice\mslice\plotting\pyplot.py", line 429, in close
    GlobalFigureManager.destroy(arg)
AttributeError: type object 'GlobalFigureManager' has no attribute 'destroy'
```
I suppose it might be better to just add another method that does what I want? But I didn't see a reason to keep a close method that doesn't work...

Reply:
Ah I see. I think the issue is that we have reimplemented the Gcf class from [`matplotlib`](https://github.com/matplotlib/matplotlib/blob/3ca813940b0b7a7640948640fe2c9ed1ebdf27e5/lib/matplotlib/_pylab_helpers.py) as https://github.com/mantidproject/mslice/blob/358_mantid_projection_plotting/mslice/plotting/globalfiguremanager.py and not provided all of the methods. Do you think we could bring our implementation into line and keep `pyplot.close` as it stands?